app/src/main/java/net/gsantner/markor/activity/ActionOrderActivity.java

```diff
@@ -0,0 +1,256 @@
+                    String key = activeKeys.get(i);
+                    _adapter.order.set(i, _keys.indexOf(key));
+                }
+                _adapter.notifyDataSetChanged();
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private void saveNewOrder() {
+        ArrayList<String> reorderedKeys = new ArrayList<>();
+        for (int i : _adapter.order) {
+            reorderedKeys.add(_keys.get(i));
+        }
+        _textActions.saveActionOrder(reorderedKeys);
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+        saveNewOrder();
+    }
+
+    @SuppressWarnings("ConstantConditions")
+    private void extractActionData() {
+        int documentType = getIntent().getExtras().getInt(EXTRA_FORMAT_KEY);
+
+        switch (documentType) {
+            default:
+            case R.id.action_format_markdown:
+                _textActions = new MarkdownTextActions(this, null);
+                break;
+            case R.id.action_format_todotxt:
+                _textActions = new TodoTxtTextActions(this, null);
+                break;
+            case R.id.action_format_plaintext:
+                _textActions = new PlaintextTextActions(this, null);
+                break;
+        }
+
+        Map<String, TextActions.ActionItem> actionMap = _textActions.getActiveActionMap();
+        _keys = new ArrayList<>(_textActions.getActionOrder());
+        _actions = new ArrayList<>();
+
+        for (String key : _keys) {
+            _actions.add(actionMap.get(key));
+        }
+    }
+
+    private class Adapter extends RecyclerView.Adapter<Holder> {
+        private List<TextActions.ActionItem> _actions;
+        private ArrayList<Integer> order;
+
+        Adapter(List<TextActions.ActionItem> actions) {
+            super();
+            _actions = actions;
+
+            order = new ArrayList<>();
+            for (int i = 0; i < _actions.size(); i++) order.add(i);
+        }
+
+        @NonNull
+        @Override
+        public Holder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+            return new Holder(getLayoutInflater().inflate(R.layout.action_order_item, parent, false));
+        }
+
+        @Override
+        public void onBindViewHolder(Holder holder, int position) {
+            holder.bindModel(_actions.get(order.get(position)));
+        }
+
+        @Override
+        public int getItemCount() {
+            return _actions.size();
+        }
+    }
+
+    private static class Holder extends RecyclerView.ViewHolder {
+        private LinearLayout _row;
+
+        Holder(View row) {
+            super(row);
+            _row = (LinearLayout) row;
+        }
+
+        void bindModel(TextActions.ActionItem action) {
+            ((ImageView) _row.getChildAt(0)).setImageResource(action.iconId);
+            ((TextView) _row.getChildAt(1)).setText(action.stringId);
+        }
+
+        void setHighlight() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/gsantner/markor/pullRequest?prid=5344500)